### DISCUSSION TITLE
Useful shell script tips for prime

### DISCUSSION THREAD

**Discussion by MauriceReeves-usds** [CDCgov/prime-reportstream#1239] [Show and tell]
So over time working on Prime I've been adding aliases and zsh functions, and I wanted to share them here. Please do the same. **Note** - I use Zsh, so you might need to edit these slightly for your specific shell.

### Aliases
```sh
export PRIME_HOME="<YOUR PATH TO>/prime-data-hub/prime-router"
alias goprime="cd $PRIME_HOME"
# this force builds the docker container and then starts up the container
alias dup="docker-compose build --force && docker-compose up"
# this does a refresh and then dup's the code
alias redup="gradle clean package && docker-compose build --force && docker-compose up"
```

### Functions
```sh
# this takes a single parameter, the path of the dependabot path
# and merges it into the current branch and then packages the code
# and spins up docker to check the file
#
# usage: depbotcheck dependabot/gradle/some/package/to/update
function depbotcheck {
        git merge origin/$1 -m "dependabot check" && gradle package && dup
}
```

**Comment by td-usds** [comment]
# See your current branch
To show your current branch name in your prompt (`${PS1}`), add this to  your `.bashrc`
```bash
git_branch() {
    git branch 2>/dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/(\1)/' | sed "s/$/ /" | sed "s/^/ /"
}

# Key item here is the call to the git_branch function at the end
# Feel free to fiddle with the colors
PS1='${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]$(git_branch)\$ '
```

Note that this is evaluated every time the `${PS1}` is rendered; so if you change branches in this repo somewhere else, then obviously the prompt will list your old branch and you'll have to hit enter to get it to update...

# git show-identity
[git-show-identity.gz](https://github.com/CDCgov/prime-reportstream/files/6670583/git-show-identity.gz)
Unpack and add this to a `+x` file called `git-show-identity` in any directory that is part of your `${PATH}`:
```bash
#!/usr/bin/env bash
NORMAL="\e[0m"
WHITE="\e[1;97m"

echo "Git Identity:"
echo -e "    Name:        ${WHITE?}$(git config --get user.name)${NORMAL?}"
echo -e "    E-Mail:      ${WHITE?}$(git config --get user.email)${NORMAL?}"
echo -e "    Signing Key: ${WHITE?}$(git config --get user.signingKey)${NORMAL?}"
```
Invoke with
```bash
git show-identity # note: no git-
```

# git identify
[git-identify.gz](https://github.com/CDCgov/prime-reportstream/files/7068767/git-identify.gz)
Place this script as `+x` in a directory that is part of your `${PATH}`. Then use it to set the **local** values for `user.name`, `user.email`, `user.signingKey` based on the information it finds in the GPG key associated with your provided identity

usage:
```bash
git identify [optional gpg-e-mail-address-here]
```
When no argument is given to identify, then it will prompt you for your GPG identity
This script will then look for the GPG identity and use it fill in the local values for `user.name`, `user.email`, `user.signingKey`. If these values are already configured, it'll ask you to confirm that you want to set the newly provided identity information.
It _assumes_ you've set `commit.gpgSign = True` yourself
